consensus/state.go

```diff
@@ -1100,6 +1103,19 @@ func (cs *State) defaultDecideProposal(height int64, round int32) {
 	} else if !cs.replayMode {
 		cs.Logger.Error("enterPropose: Error signing proposal", "height", height, "round", round, "err", err)
 	}
+
+	// post data to ipfs
+	// TODO(evan): don't hard code context and timeout
+	if cs.IpfsAPI != nil {
+		// longer timeouts result in block proposers failing to propose blocks in time.
+		ctx, cancel := context.WithTimeout(context.Background(), time.Millisecond*1500)
+		// TODO: post data to IPFS in a goroutine
+		err := block.PutBlock(ctx, cs.IpfsAPI.Dag().Pinning())
+		if err != nil {
+			cs.Logger.Error(fmt.Sprintf("failure to post block data to IPFS: %s", err.Error()))
+		}
+		cancel()
```

Comment:
Nit: It seem more idiomatic to use `defer cancel()` directly under the var defintion.

Reply:
resolved with a defer https://github.com/lazyledger/lazyledger-core/pull/178/commits/52fdd1b1328382ff1d2ed4b0c807b544f83ede3c